exe/facter.cc

```diff
@@ -299,6 +281,9 @@ int main(int argc, char **argv)
             log(level::warning, "Facter was called recursively, skipping external facts. Add '--no-external-facts' to silence this warning");
           } else {
             environment::set("INSIDE_FACTER", "true");
+            if (vm.count("external-dir")) {
+                external_directories.push_back(vm["external-dir"].as<string>());
```

Comment:
I'm going to go with `string` for now and add the `move`. It makes more sense to me given the description of the option (which matches online docs I found for Facter, which say you can only specify one directory with this option).